extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -37,6 +38,83 @@ class AmpStickyAd extends AMP.BaseElement {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+
+    const children = this.getRealChildren();
+    user.assert((children.length == 1 && children[0].tagName == 'AMP-AD'),
+        'amp-sticky-ad must have a single amp-ad child');
+    /** @const @private {!Element} */
+    this.ad_ = children[0];
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = this.getViewport();
+
+    /**
+     * On viewport scroll, check requirements for amp-stick-ad to display.
+     * @const @private {!Unlisten}
+     */
+    this.scrollUnlisten_ =
+        this.viewport_.onScroll(() => this.displayAfterScroll_());
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
+    if (!this.isExperimentOn_) {
+      dev.warn(TAG, `TAG ${TAG} disabled`);
+      return Promise.resolve();
+    }
+    return Promise.resolve();
+  }
+
+  /** @override */
+  detachedCallback() {
+    this.removeOnScrollListener();
+  }
+
+  /**
+   * The function that remove listener to viewport onScroll event.
+   * @private
+   */
+  removeOnScrollListener() {
+    this.scrollUnlisten_();
+    this.scrollUnlisten_ = null;
+  }
+
+  /**
+   * The listener function that listen on onScroll event and
+   * show sticky ad when user scroll at least one viewport and
+   * there is at least one more viewport available.
+   * @private
+   */
+   // TODO(zhouyx): not really a member variable, need to move
+  displayAfterScroll_() {
+    this.scrollTop_ = this.viewport_.getScrollTop();
+    this.viewportHeight_ = this.viewport_.getSize().height;
+    // TODO(zhouyx): When calculate 'has scrolled through at least 1 viewport'
+    // do we want to count height from top or from initial position?
+    // Will assume user start from `this.scrollTop_ = 0` here. Need
+    // to figure out later.
+
+    // Check user has scrolled at least one viewport from init position.
+    if (this.scrollTop_ > this.viewportHeight_) {
+      this.scrollHeight_ = this.viewport_.getScrollHeight();
+      const remainHeight = this.scrollHeight_
+          - this.scrollTop_ - this.viewportHeight_;
+      if (remainHeight < this.viewportHeight_) {
```

Comment:
Please correct me if I'm wrong, but it sounds like the actual condition is that `if (scrollHeight < viewportHeight * 2) {disconnect}`, right? If so then, yes, definitely we can put it outside of this `if()`